streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1929,10 +1930,18 @@ public synchronized ClientInstanceIds clientInstanceIds(final Duration timeout)
 
             // could be `null` if telemetry is disabled on the consumer itself
             if (instanceId != null) {
-                clientInstanceIds.addConsumerInstanceId(
-                    clientFuture.getKey(),
-                    instanceId
-                );
+                final String clientFutureKey = clientFuture.getKey();
+                if (clientFutureKey.toLowerCase(Locale.getDefault()).contains("producer")) {
```

Comment:
Should we look for `.endsWith("-producer")` instead of using `contains()`?

Reply:
I was aiming for something more generic if the thread names were to change, but that's unlikely, so I'm updating it.